blog/content/post/2020-01-15-GRUB2_and_3mdeb_minisummit.md

```diff
@@ -0,0 +1,177 @@
+relation and organize meeting during which 3mdeb engineers and team leaders
+presented GRUB related topics. This blog post summarize discussion and create
+reference point for interested parties:
+
+Somehow "minisummit" name suvived since no one offered btter title after
+publishing [Qubes OS and 3mdeb 'minisummit' 2019](https://blog.3mdeb.com/2019/2019-08-07-qubes-os-and-3mdeb-minisummit/)
+blog post.
+
+What we discussed during our meeting with Daniel:
+
+* [Intro](https://cloud.3mdeb.com/index.php/s/Dk526gCtpjXRSDK) - introduction,
+  motivation and agenda presented by me
+* [Redundant GRUB2 env file](https://cloud.3mdeb.com/index.php/s/bBcbXNkHwLBPZLn) - Maciej
+  Pijanowski presented redundant GRUB2 environment file,
+  feature needed for power fail safe upgrades, similar mechanism was already
+  implemented in U-Boot
+* [TPM support in GRUB2 for legacy boot mode](https://cloud.3mdeb.com/index.php/s/gxg595WG35xSjjb)
+  - update on [talk presented on LPC 2019](https://linuxplumbersconf.org/event/4/contributions/517/)
+  in which Michał Żygowski highlight changes made in TrustedGRUB2 and build
+  base for discussion about merging changes to GRUB2 mainline
+* [GRUB2 security features overview](https://cloud.3mdeb.com/index.php/s/trSb3RnjfJWxkM3) - general talk
+  overviewing security features in GRUB2 presented by me
+* [Python 3 support in GRUB2](https://cloud.3mdeb.com/index.php/s/7KKJ5cQfGxPkYyi) - talk in which Michał Żygowski present how we
+  used Python with GRUB2 and start discussion about making Python 3 first class
+  citizen in GRUB2 bootloader
+* [AMD TrenchBoot support in GRUB2](https://cloud.3mdeb.com/index.php/s/SjLdXJomaS6obYH) - final talk in
+  which I present status of AMD TrenchBoot support, implementation and possible paths to
+  upstream related code
+
+We also published all videos on our [YouTube channel]().
+
+# Redundant GRUB2 env file
+
+Key problem mentioned during presentations are lack of integrity check in
+current implementation of GRUB2, what means any corruption of environment file
+in GRUB may lead to boot failure that can be recovered only by manual
+intervention. Since in most cases GRUB2 is used as bootloader in general
+purpose computers this was not a problem. Even if in reality it happen people
+know how to recover either through GRUB command line or live booting Linux
+distro.
+
+Unfortunately, if trying to use GRUB2 in embedded environment there is risk of
+ending up with unbootable product, which of course leads to all other problems
+on vendor side.
+
+We all agreed that this feature is needed in GRUB2 and if 3mdeb will get
+correct sponsoring from vendors using GRUB2 such support will be developed and
+contributed. On the other hand some vendors assume that file corruption is so
+rare that they don't want to spend additional engineering hours on such
+support.
+
+Before such support is available there are still methods to mitigate impact of
```

Comment:
Before such support is available, there are still methods to mitigate the impact of the problem for which reference you can find in the presentation.